tests/diagnostics/uninit-var.slang

```diff
@@ -0,0 +1,35 @@
+// uninit-var.slang
```

Comment:
There likely should be isolated tests, each testing for a different uninitialized variable warning/error.

The 3 warning/errors are attached below:
```
DIAGNOSTIC(41015, Error, usingUninitializedValue, "use of uninitialized value '$0'")
DIAGNOSTIC(41016, Warning, returningWithUninitializedOut, "returning without initializing out parameter '$0'")
DIAGNOSTIC(41017, Warning, returningWithPartiallyUninitializedOut, "returning without fully initializing out parameter '$0'")
```

Reply:
Even simple tests produce more than one of these diagnostics, so I am not sure if we can isolate them into separate tests. May be I can isolate 41015 + 41016 into one, and 41015 + 40107 into one.
Also, should I also change 41015 into a Warning?